src/main/controllers/crash-report-service.js

```diff
@@ -0,0 +1,30 @@
+const { init } = require('@sentry/electron');
+const guideSettings = require('./models/guide-setting')();
+const log = require('electron-log');
+// DSN will be inject during build time
+const sentry = require('../assets/data/sentry');
+let hasStarted = false;
+
+module.exports = {
+	startCrashReport: async () => {
+		if (process.env.NODE_ENV !== 'development' && process.env.MODE !== 'test') {
+			const result = await guideSettings.findAll();
+			if (result[0].crashReportAgreement && !hasStarted) {
```

Comment:
This will work most of the time, but Sentry also offers the `shouldSend` callback which is more reliable and cleaner. I know you've added the "startCrashReport" call as a side effect when updating the settings, but that violates the SRP (updating the database shouldn't start a service) and tightly couples the settings handler and our Sentry implementation.

https://docs.sentry.io/clients/electron/config/#options